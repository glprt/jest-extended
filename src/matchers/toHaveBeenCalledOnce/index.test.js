import matcher from './';

expect.extend(matcher);

describe('.toHaveBeenCalledOnce', () => {
  test('should fail when given a mock that has not been called', () => {
    const mock = jest.fn();
    expect(() => expect(mock).toHaveBeenCalledOnce()).toThrowErrorMatchingSnapshot();
  });

  test('should fail when given a mock that has been called more than once', () => {
    const mock = jest.fn();
    mock(1);
    mock(2);
    expect(() => expect(mock).toHaveBeenCalledOnce()).toThrowErrorMatchingSnapshot();
  });

  test('should pass when given a mock that has been called exactly once', () => {
    const mock = jest.fn();
    mock(1);
    expect(mock).toHaveBeenCalledOnce();
  });
});

describe('.not.toHaveBeenCalledOnce', () => {
  test('should pass when given a mock that has not been called', () => {
    const mock = jest.fn();
    expect(mock).not.toHaveBeenCalledOnce();
  });

  test('should pass when given a mock that has been called more than once', () => {
    const mock = jest.fn();
    mock(1);
    mock(2);
    expect(mock).not.toHaveBeenCalledOnce();
  });

  test('should fail when given a mock that has been called exactly once', () => {
    const mock = jest.fn();
    mock(1);
    expect(() => expect(mock).not.toHaveBeenCalledOnce()).toThrowErrorMatchingSnapshot();
  });
});
