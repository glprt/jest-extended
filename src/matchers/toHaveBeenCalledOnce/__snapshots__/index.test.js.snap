// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveBeenCalledOnce should fail when given a mock that has been called exactly once 1`] = `
"<dim>expect(</><red>received</><dim>).not.toHaveBeenCalledOnce()</>

Expected mock to not have been called once, mock called with:
  <red>[[1]]</>
"
`;

exports[`.toHaveBeenCalledOnce should fail when given a mock that has been called more than once 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce()</>

Expected mock to have been called once, mock called with:
  <red>[[1], [2]]</>
"
`;

exports[`.toHaveBeenCalledOnce should fail when given a mock that has not been called 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce()</>

Expected mock to have been called once, mock called with:
  <red>[]</>
"
`;
